tests/api/test_pagination.py

```diff
@@ -0,0 +1,1654 @@
+                product_type=product_types[0],
+                category=category2,
+                is_published=True,
+            ),
+            Product(
+                name="Test product",
+                slug="product",
+                price=Money("10.00", "USD"),
+                minimal_variant_price_amount=Decimal("10.00"),
+                product_type=product_types[0],
+                category=category3,
+                is_published=True,
+            ),
+        ]
+    )
+
+
+def create_global_cursor(instance, fields_list):
+    return to_global_cursor([get_field_value(instance, field) for field in fields_list])
+
+
+def create_cursor_from_user_email(email, fields_list):
+    user = (
+        User.objects.filter(email=email)
+        .annotate(order_count=Count("orders__id"))
+        .first()
+    )
+    return create_global_cursor(user, fields_list)
+
+
+def create_cursor_from_group_name(name, fields_list):
+    group = auth_models.Group.objects.get(name=name)
+    return create_global_cursor(group, fields_list)
+
+
+def create_cursor_from_service_account_name(name, fields_list):
+    service_account = ServiceAccount.objects.get(name=name)
+    return create_global_cursor(service_account, fields_list)
+
+
+def create_cursor_from_checkout_date(token, fields_list):
+    checkout = Checkout.objects.get(token=token)
+    return create_global_cursor(checkout, fields_list)
+
+
+def create_cursor_from_sales_name(name, fields_list):
+    sale = Sale.objects.get(name=name)
+    return create_global_cursor(sale, fields_list)
+
+
+def create_cursor_from_voucher_code(code, fields_list):
+    voucher = Voucher.objects.get(code=code)
+    return create_global_cursor(voucher, fields_list)
+
+
+def create_cursor_from_menu_name(name, fields_list):
+    menu = (
+        Menu.objects.filter(name=name).annotate(items_count=Count("items__id")).first()
+    )
+    return create_global_cursor(menu, fields_list)
+
+
+def create_cursor_from_order_token(token, fields_list):
+    order_qs = Order.objects.filter(token=token)
+    order = OrderSortField.qs_with_payment(order_qs).first()
+    return create_global_cursor(order, fields_list)
+
+
+def create_cursor_from_page_slug(slug, fields_list):
+    page = Page.objects.get(slug=slug)
+    return create_global_cursor(page, fields_list)
+
+
+def create_cursor_from_payment_token(token, fields_list):
+    payment = Payment.objects.get(token=token)
+    return create_global_cursor(payment, fields_list)
+
+
+def create_cursor_from_attribute_slug(slug, fields_list):
+    attribute = Attribute.objects.get(slug=slug)
+    return create_global_cursor(attribute, fields_list)
+
+
+def create_cursor_from_category_slug(slug, fields_list):
+    category = Category.objects.filter(slug=slug)
+    category = CategorySortField.qs_with_product_count(category)
+    category = CategorySortField.qs_with_subcategory_count(category).first()
+
+    return create_global_cursor(category, fields_list)
```

Comment:
Why do we need those? Why not use the API to fetch a page of 2 items and obtain the cursor to the next page and so on? We don't need to document how cursors are created as only the GraphQL server will ever create them and we should be able to change their format as often as we see fit.